Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java

```diff
@@ -0,0 +1,73 @@
+package xmlteam4.Project.services;
+
+import org.apache.xpath.operations.Bool;
+import org.exist.http.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import xmlteam4.Project.repositories.ScientificPaperRepository;
+import xmlteam4.Project.utilities.dom.DOMParser;
+import xmlteam4.Project.utilities.idgenerator.IDGenerator;
+import xmlteam4.Project.utilities.transformer.DocumentXMLTransformer;
+
+@Service
+public class ScientificPaperService {
+
+    @Autowired
+    private ScientificPaperRepository scientificPaperRepository;
+
+    @Autowired
+    private ScientificPaperService scientificPaperService;
+
+    //@Autowired
+    //private XSLTransfomer xslTransfomer;
+
+    @Autowired
+    private DOMParser domParser;
+
+    @Autowired
+    private DocumentXMLTransformer documentXMLTransformer;
+
+    @Value("${scientific-paper-schema-path}")
+    private String scientificPaperSchemaPath;
+
+    public String findOne(String id) throws Exception {
+        String paper = scientificPaperService.findOne(id);
+        if(paper == null) {
+            throw new NotFoundException(String.format("Scientific paper with id %s is not found", id));
+        }
+
+        return paper;
+
+    }
+
+    public String create(String xml) throws Exception {
+        Document document = domParser.buildDocument(xml, scientificPaperSchemaPath);
+
+        String id = IDGenerator.createID();
+        document.getDocumentElement().setAttribute("id", id);
```

Comment:
After generating an ID for scientific paper, you should generate id for every child element which can have an ID(read schema for more info). ID should be URL and generation can be done using sequencer. Check other document types for the same issue.

Reply:
ID's have been generated for all child elements and ID is now an URL